### PR TITLE
<foo/> and testing framework updates

### PR DESCRIPTION
**Pull request by billstclair**
I moved my pull request from a pass-through on myrho's fork to my own direct fork. This fixes four things:

`<foo/>` now parses correctly. Before it needed attributes or some space before the closing />.

Updated to the latest elm-test, which no longer requires `Main.elm` or the `rtfeldman/node-test-runner` package.

Updated to `elm-verify-examples` and its new doc test syntax.

Added `sysconfcpus` to `.travis.xml` for epic Elm compile time improvement, minutes become seconds.

Trimmed the giant 1000-element S3 directory string to 100 elements, to prevent timeouts on Travis.

### CODE REVIEW

File: src/Xml/Decode.elm [hunk 1]
```diff
@@ -97,8 +97,14 @@ parseSlice first firstClose trimmed =
             [] ->
                 if String.startsWith "?" tagName then
                     Ok ( DocType tagName props, firstClose + 1 )
-                else if (String.contains "/>" trimmed) then
-                    Ok ( Tag tagName props (Object []), firstClose + 1 )
+                else if String.endsWith "/" beforeClose then
+                    let
+                        tag = if String.endsWith "/" tagName then
```
Comment (eeue56) [inline src/Xml/Decode.elm:102]: the `if` should start on a newline ideally but no big deal

Comment (billstclair) [reply src/Xml/Decode.elm:102]: I've started using elm-format in my editor, but when I tried that on one of your source files, it changed it significantly, so I'm not using it there. And I haven't internalized all the rules.

Would you like me to push a little change to add the newline?

**Comment by eeue56** [comment]
Published as 2.2.3

**Comment by billstclair** [comment]
Yay! How do you like the 2 minute and 6 second Linux testing?

I don't know what's up with Travis macOS testing. It's been very slow to start of late.

**Comment by eeue56** [comment]
I'm well aware of the libsysconfig trick and have looked into the causes :)

**Comment by billstclair** [comment]
Not surprising. I'm a lot newer at Elm than you, so it was a revelation to me a few days ago.

**Comment by eeue56** [comment]
The next release of Elm should be "fixed" of this problem at least

**Comment by billstclair** [comment]
Glad to hear it. Have I said recently that Elm rocks! Well, Elm rocks! In spades!

The crypto code that I've written recently is the most beautiful code I've ever written in my 40 years of writing software for a living. And Elm is largely responsible for that.

**Pull request merged**